modules/packages/BLAS.chpl

```diff
@@ -641,7 +640,7 @@ module BLAS {
         n = Bdom.dim(2).size : c_int;
 
     if m != n then
-      try! halt("Non-square array of dimensions %ix%i passed to trmm".format(m, n));
+      throw new IllegalArgumentError("B", "Non-square array of dimensions %ix%i".format(m, n));
```

Comment:
There are many of these that include the method name (trmm in this case). Should we include that in the new error? I'd presume it'd help, until we have stack traces on errors...

Reply:
I forgot that traces aren't on by default, so I've restored the original error messages.